src/foam/dao/AbstractDAO.js

```diff
@@ -583,16 +583,32 @@ this.select_(x, new RemoveSink(x, this), skip, limit, order, predicate);
         },
       ],
       javaCode: `
-if ( ( limit > 0 ) && ( limit < AbstractDAO.MAX_SAFE_INTEGER ) ) {
-  sink = new LimitedSink(limit, 0, sink);
+var innerSink = sink;
+while ( innerSink instanceof ProxySink ) {
+  innerSink = ((ProxySink) innerSink).getDelegate();
 }
 
-if ( ( skip > 0 ) && ( skip < AbstractDAO.MAX_SAFE_INTEGER ) ) {
-  sink = new SkipSink(skip, 0, sink);
-}
+// REVIEW: limit, skip may not be compatible with select(COUNT), eg.
+// Are the below queries valid use cases we needs to support?
+// 1) dao.limit(5).select(COUNT) => return the number of record or 5 when there are more records
+// 2) dao.skip(5).select(COUNT) => return the number of record - 5
+//
```

Comment:
I think COUNT should be applied to the result of the skip or limit. 
limit(5) with predicate which filters to 2 would return count 2. 
but if a use of COUNT is show total in a dao from the UI -   20 of 1million, then count has to apply before any other operator - where, skip, limit, .... 
so either always before or always after.

Reply:
The predicate sink is already before limit, if the predicate filters to 2 records then the count will be 2 either with and without limit(5). I think it does help when there are more data and the limit sink could early detach and return.